xcube/core/store/descriptor.py

```diff
@@ -139,17 +130,20 @@ def __init__(self,
         self.data_vars = list(data_vars) if data_vars else None
         self.attrs = dict(attrs) if attrs else None
 
+    def _assert_type_id(self, type_id: str):
+        if not TYPE_ID_DATASET.is_compatible(type_id):
+            raise ValueError('TypeId must be "dataset" type id')
```

Comment:
```suggestion
            raise ValueError('type_id must be compatible with "dataset"')
```

Note that value errors are used to signal developer errors, hence it should include the name of the actual variable that includes the invalid value (`type_id` here).